sequencer/batchbuilder.go

```diff
@@ -333,20 +333,17 @@ func (s *Sequencer) shouldCloseSequenceInProgress(ctx context.Context) bool {
 		}
 	}
 	// Check if it has been too long since a batch is virtualized
-	lastBatchTime, err := s.state.GetLastBatchTime(ctx, nil)
-	if err != nil && !errors.Is(err, state.ErrNotFound) {
-		log.Errorf("failed to get last batch time, err: %v", err)
+	isPreviousBatchVirtualized, err := s.state.IsBatchVirtualized(ctx, s.lastBatchNum-1, nil)
+	if err != nil {
+		log.Errorf("failed to get last virtual batch num, err: %v", err)
 		return false
 	}
-	if lastBatchTime.Before(time.Now().Add(-s.cfg.LastTimeBatchMaxWaitPeriod.Duration)) && len(s.sequenceInProgress.Txs) > 0 {
-		isProfitable := s.isSequenceProfitable(ctx)
-		if isProfitable {
-			log.Info(
-				"current sequence should be closed because LastTimeBatchMaxWaitPeriod has been exceeded, " +
-					"there are pending sequences to be sent and they are profitable")
-			return true
-		}
+	if time.Unix(s.sequenceInProgress.Timestamp, 0).Add(s.cfg.MaxTimeForBatchToBeOpen.Duration).After(time.Now()) &&
```

Comment:
```suggestion
	if time.Unix(s.sequenceInProgress.Timestamp, 0).Add(s.cfg.MaxTimeForBatchToBeOpen.Duration).Before(time.Now()) &&
```

If I understand it well, it should be before: `time to close = sequence in progress timestamp + max open time`.
Let's say `time to close = 10:00`, then:

- if the current time is 10:30, we should close, because we should have closed 30 mins ago already! Note that in that case 10:00 is before 10:30
- If the current time is 9:30, we shouldn't close as we have 30 more minutes to reach the time to close. In this case 10:00 is after 9:30, or what is the same is not before